release-notes/source/source-code.rst

```diff
@@ -13,6 +13,49 @@ The following tree shows the SoC hierarchy:
    :scale: 60%
    :align: center
 
+.. _machine-overrides-extender:
+
+What is machine-overrides-extender
+==================================
+
+The `Machine Overrides Extender class <https://github.com/Freescale/meta-freescale/blob/master/classes/machine-overrides-extender.bbclass>`_
+is a class from **meta-freescale** created to extend some machine overrides
+depending on which `SoC <https://en.wikipedia.org/wiki/System_on_a_chip>`_
+is part of a
+`MPSoC <https://en.wikipedia.org/wiki/Multiprocessor_system_on_a_chip>`_.
+For example, a MPSoC has IPU, and GPU for 3D, but does not have GPU for 2D.
+
+The list of SoC on a given MPSoC is known and does not change overtime. The
+current MPSoC configuration can be found in file
+`imx-base.inc <https://github.com/Freescale/meta-freescale/blob/master/conf/machine/include/imx-base.inc>`_,
+the lines where the variable ``MACHINEOVERRIDES_EXTENDER`` is set for each MPSoC.
+When a new machine file is added to **meta-freescale**, defining the MPSoC is
+enough.
+
+The Machine Overrides Extender class adds the pre-determined list of SoC
+configured for that MPSoC to the ``MACHINEOVERRIDE``, and as a consequence,
+the BSP packages are properly configured for that machine by default.
+
+In other words, when adding a new machine support, the user only need to know
+the MPSoC for that machine, the **meta-freescale** BSP support is already
+configured to provide the proper package depending on the MPSoC (and this is
+configured via Machine Overrides Extender).
+
+This meas any new machine added in future can use the configured BSP provided by
+**meta-freescale**.
+
+On top of the hardware architecture which impacts the BSP, there are also the
+software architecture. There are three supported BSPs:
+
+* ``imx-generic-bsp``: Used to configure any generic aspect of the BSP, for example,
+  those configurations that depends only on the MPSoC.
+
+* ``imx-mainline-bsp``: Used to configure the mainline packages.
+
+* ``imx-nxp-bsp``: Used to configure the packages provided by NXP.
+
+The BPS overrides are provided to allow the user to chose between the BSPs and
+using the proper package for each case.
```

Comment:
I believe an example using mx8m or similar would be good to add as it shows more details about the process.

Reply:
@otavio an example on how to add a new machine? Or an example on how to switch from one BSP to the other?

Reply:
@otavio , please take a look in the example. My concern is to add a sample code that makes sense now, but will be deprecated in near future, but if you think there is something else to add, please, let me know.